### PR TITLE
DDCE-3683: cleanup url builder + update deps

### PR DESCRIPTION
**Pull request by frikit**

### CODE REVIEW

File: app/controllers/BaseController.scala [hunk 1]
```diff
@@ -49,9 +47,6 @@ abstract class BaseController @Inject() (cc: MessagesControllerComponents)(impli
 
   lazy val ggSignInRedirect: Result = toGGLogin(loginContinue)
 
-  def logoutLink(implicit request: Request[_]): Html =
```
Comment (frikit) [inline app/controllers/BaseController.scala:52]: not used anywhere

File: app/metrics/HasMetrics.scala [hunk 2]
```diff
@@ -1,57 +0,0 @@
...
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package metrics
-
-import com.codahale.metrics._
-import com.kenshoo.play.metrics.Metrics
-
-trait HasMetrics {
```
Comment (frikit) [inline app/metrics/HasMetrics.scala:22]: not used anywhere

File: app/views/error_template.scala.html [hunk 3]
```diff
@@ -37,5 +36,4 @@ <h1 class="govuk-heading-xl" id="error-heading">@heading</h1>
 
 @govukWrapper(PageTitle(pageTitle)) {
     @mainContent
-    @sidebarLink.map{link=>@link}
```
Comment (frikit) [inline app/views/error_template.scala.html:40]: it is always none , only usage was in test and I deleted that test as well

**Review by rreddy80 — APPROVED** [review]
Some issues with smoke (cause of DES data errors) and perf tests, cause of some external urls

**Pull request merged**